### PR TITLE
Socks5 example

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
Error

```
Exception in thread "main" java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:283)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:309)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:350)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:803)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:976)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:971)
	at sockslib.utils.StreamUtil.read(StreamUtil.java:42)
	at sockslib.client.GenericSocksMethodRequester.doRequest(GenericSocksMethodRequester.java:66)
	at sockslib.client.Socks5.buildConnection(Socks5.java:200)
	at sockslib.client.SocksSocket.connect(SocksSocket.java:208)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.kt:138)
	at okhttp3.internal.connection.ConnectPlan.connectSocket(ConnectPlan.kt:265)
	at okhttp3.internal.connection.ConnectPlan.connectTcp(ConnectPlan.kt:136)
	at okhttp3.internal.connection.FastFallbackExchangeFinder$launchTcpConnect$1.runOnce(FastFallbackExchangeFinder.kt:138)
```

File: samples/guide/build.gradle.kts [hunk 1]
```diff
@@ -8,10 +8,15 @@ dependencies {
   implementation(projects.mockwebserver)
   implementation(projects.okhttpTestingSupport)
   implementation(projects.okhttpTls)
+  implementation(projects.loggingInterceptor)
   implementation(libs.animalsniffer.annotations)
   implementation(libs.squareup.moshi)
   implementation(libs.squareup.okio.fakefilesystem)
   kapt(libs.squareup.moshi.compiler)
+
+  implementation("org.testcontainers:testcontainers:1.17.6")
```
Comment (yschimke) [inline samples/guide/build.gradle.kts:17]: TODO: move to version catalog

File: samples/guide/build.gradle.kts [hunk 2]
```diff
@@ -8,10 +8,15 @@ dependencies {
...
+  implementation("com.github.mike10004:fengyouchao-sockslib:1.0.6")
```
Comment (swankjesse) [inline samples/guide/build.gradle.kts:18]: What’s with the group name on this? Shouldn’t it be https://github.com/fengyouchao/sockslib ?

(I’m anxious about the implicit endorsement of this library)

Comment (swankjesse) [reply samples/guide/build.gradle.kts:18]: It’s kind of kitchen-sinky

https://github.com/fengyouchao/sockslib/blob/master/src/main/java/sockslib/utils/mongo/MongoDBUtil.java

**Comment by swankjesse** [comment]
I think it’s a good example but I’m reluctant to add it to the library itself.

Now that we’ve got Renovate pushing us to keep all of our dependencies I’m more aware of just how many unrelated dependencies OkHttp has. I’d like to get that down! For example, I don’t like that we have all the other clients in `samples/compare`.

Would this be better posted elsewhere? A blog post or something?

Also I think Socks5 is a means and not an end. I think I’d prefer a sample of how to hook up OkHttp to Tor or Docker or something.

**Comment by yschimke** [comment]
Ok, I might look at the Tor options instead.

File: samples/guide/build.gradle.kts (hunk 2 above)
Comment (yschimke) [reply samples/guide/build.gradle.kts:18]: It's not officially published.

**Pull request closed without merging**